actors/evm/src/lib.rs

```diff
@@ -53,16 +51,32 @@ impl EvmContractActor {
         RT: Runtime<BS>,
     {
         // TODO ideally we would be checking that we are constructed by the EAM actor,
-        //   so instead we check for init and then assert that we have a delegated address.
-        rt.validate_immediate_caller_type(iter::once(&Type::Init))?;
-
-        // BIG LARGE TODO AS IN TODO TOMORROW
-        // - @mriise
-        // if rt.lookup_address(rt.message().caller().id().unwrap()).is_none() {
-        //     return Err(ActorError::assertion_failed(
-        //         "EVM actor created without a delegated address".into(),
-        //     ));
-        // }
+        //   but instead we check for init and then assert that we have a delegated address.
+        // rt.validate_immediate_caller_is(vec![&EAM_ACTOR_ADDR])?;
+        rt.validate_immediate_caller_type(vec![&Type::Init])?;
+
+        let delegated_addr = rt.lookup_address(rt.message().receiver().id().unwrap());
+        if delegated_addr.is_none() {
+            return Err(ActorError::assertion_failed(format!(
+                "EVM actor {} created without a delegated address",
+                rt.message().receiver()
+            )));
+        } else if delegated_addr
+            .and_then(|a| match a.payload() {
+                Payload::Delegated(d) => (d.namespace() == EAM_ACTOR_ID as u64).then_some(()),
+                _ => None,
+            })
+            .is_none()
+        {
+            return Err(ActorError::assertion_failed(format!(
+                "EVM actor {} created without a delegated address from the EAM",
+                delegated_addr.unwrap()
+            )));
+        }
```

Comment:
This is a bit of a strange way to do this (handling of the option).

Also, we lookup the receivers "ethereum" address below. We should be able to just check "is my address a 'real' ethereum address". See line 101/102.